compiler/hie-core/src/Development/IDE/Core/Rules.hs

```diff
@@ -142,10 +146,18 @@ writeIfacesAndHie ifDir main =
             createDirectoryIfMissing True (takeDirectory fp)
             let ifaceFp = replaceExtension fp ".hi"
             let hieFp = replaceExtension fp ".hie"
-            writeIfaceFile
-                (hsc_dflags session)
-                ifaceFp
-                (hm_iface $ tmrModInfo tcm)
+            let iface0 = hm_iface $ tmrModInfo tcm
+            let iface =
```

Comment:
imho, this would be easier to read if you just case on `mbPkgName`, i.e.
```
let iface = case mbPkgName of
        Nothing -> iface0
        Just pkgName -> iface0 {mi_module = mi_module iface0 {moduleUnitId = stringToUnitId pkgName}}
```

Reply:
made it a bit cleaner.